### PR TITLE
Update RELEASES.md for 1.44.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/rust/blob/relnotes-1.44.0/RELEASES.md)

r? @Mark-Simulacrum 
cc @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,134 @@
...
+
+- [Expansion-driven outline module parsing][69838]
+
+These are still rejected semantically, so you will likely receive an error but
+these changes can be seen and parsed by macros and conditional compilation.
+
+Compiler
+--------
+- [Rustc now respects the `-C codegen-units` flag in incremental mode.][70156]
+  Additionally when incremental mode rustc defaults to 256 codegen units.
```
Comment (jonas-schievink) [inline RELEASES.md:19]:
```suggestion
  Additionally when in incremental mode rustc defaults to 256 codegen units.
```

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,134 @@
...
+[`PathBuf::capacity`]: https://doc.rust-lang.org/beta/std/path/struct.PathBuf.html#method.capacity
+[`PathBuf::clear`]: https://doc.rust-lang.org/beta/std/path/struct.PathBuf.html#method.clear
+[`PathBuf::reserve`]: https://doc.rust-lang.org/beta/std/path/struct.PathBuf.html#method.reserve
+[`PathBuf::reserve_exact`]: https://doc.rust-lang.org/beta/std/path/struct.PathBuf.html#method.reserve_exact
+[`PathBuf::shrink_to_fit`]: https://doc.rust-lang.org/beta/std/path/struct.PathBuf.html#method.shrink_to_fit
+[`f32::to_int_unchecked`]: https://doc.rust-lang.org/beta/std/primitive.f32.html#method.to_int_unchecked
+[`f64::to_int_unchecked`]: https://doc.rust-lang.org/beta/std/primitive.f64.html#method.to_int_unchecked
+[`Layout::align_to`]: https://doc.rust-lang.org/beta/std/alloc/struct.Layout.html#method.align_to
+[`Layout::pad_to_align`]: https://doc.rust-lang.org/beta/std/alloc/struct.Layout.html#method.pad_to_align
+[`Layout::array`]: https://doc.rust-lang.org/beta/std/alloc/struct.Layout.html#method.array
```
Comment (tspiteri) [inline RELEASES.md:129]: `Layout::extend` is missing. #69362

**Review by tspiteri — COMMENTED** [review]
Also, the following PRs are labelled relnotes in the 1.44 milestone but I didn't find them.
* #68899
* #68191
* #66059
* #69373

**Review by Mark-Simulacrum — COMMENTED** [review]
Gah I think I typed this up a week or so back but forgot to actually submit it.

**Comment by steveklabnik** [comment]
Just want to raise that the release is next week, so getting this in sometime over the next seven days would be great!

**Comment by XAMPPRocky** [comment]
I plan to finish this by Sunday.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,134 @@
...
+
+Compatibility Notes
+-------------------
+- [Rustc now correctly generates static libraries on Windows GNU targets with
+  the `.a` extension, rather than the previous `.lib`.][70937]
+- [Removed the `-C no_integrated_as` flag from rustc.][70345]
+- [The `file_name` property in JSON output of macro errors now points the actual
+  source file rather than the previous format of `<NAME macros>`.][70969]
+  **Note:** this may not point a file that actually exists on the user's system.
+- [The minimum required external LLVM version has been bumped to LLVM 8.][71147]
```
[the remark this answers is not in the document]
Comment (XAMPPRocky) [reply RELEASES.md:101]: What is the compatibility note? I'm not aware of the LLVM 8 change.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,156 @@
...
+- [`convert::Infallible` now implements `Hash`.][70281]
+- [`OsString` now implements `DerefMut` and `IndexMut` returning
+  a `&mut OsStr`.][70048]
+- [Unicode 13 is now supported.][69929]
+- [`String` now implements `From<&mut str>`.][69661]
+- [`IoSlice` now implements `Copy`.][69403]
+- [`Vec<T>` now implements `From<[T; N]>`.][68692] Where `N` is less than 32.
+- [`proc_macro::LexError` now implements `fmt::Display` and `Error`.][68899]
+- [`from_le_bytes`, `to_le_bytes`, `from_be_bytes`, and `to_be_bytes` methods are
+  now `const` for all integer types.][69373]
```
Comment (tspiteri) [inline RELEASES.md:49]: `from_ne_bytes` and `to_ne_bytes` are missing.

**Comment by Mark-Simulacrum** [comment]
@bors r+ rollup=always

Not nominating for beta/stable backport as we always cherry pick this anyway.

**Pull request merged**